src/llm/openai.rs

```diff
@@ -33,6 +41,39 @@ pub struct Response {
     choices: Vec<Choice>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct OpenAIEmbeddingResponse {
+    id: String,
+    object: String,
+    model: String,
+    data: Embedding,
+    usage: Usage,
+}
```

Comment:
Check out https://platform.openai.com/docs/guides/embeddings for the correct contract!